lib/services/project-changes-service.ts

```diff
@@ -219,6 +219,12 @@ export class ProjectChangesService implements IProjectChangesService {
 	}
 
 	private containsNewerFiles(dir: string, skipDir: string, projectData: IProjectData, processFunc?: (filePath: string, projectData: IProjectData) => boolean): boolean {
+
+		// ignore directories and files ending with .something (.bin, .git ...)
+		if (/.*[/\\]\.[^/\\]*$/.test(dir)) {
```

Comment:
While this code is correct I would prefer a non-regex approach. For example:
```TypeScript
const dirName = path.basename(dir);
if (_.startsWith(dirName, '.')) {
    return false;
}
```

I think that a bit more readable.